include/dt-bindings/gpio/gpio.h

```diff
@@ -6,102 +6,177 @@
 #ifndef ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_GPIO_H_
 #define ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_GPIO_H_
 
-
-/**
- * @name GPIO direction flags
- * The `GPIO_DIR_*` flags are used with `gpio_pin_configure` or `gpio_port_configure`,
- * to specify whether a GPIO pin will be used for input or output.
- * @{
+/** GPIO pin is active (has logical value '1') in low state. */
+#define GPIO_ACTIVE_LOW         (1 << 0)
+/** GPIO pin is active (has logical value '1') in high state. */
+#define GPIO_ACTIVE_HIGH        (0 << 0)
+
+/** Configures GPIO output in single-ended mode (open drain or open source). */
+#define GPIO_SINGLE_ENDED       (1 << 1)
+/** Configures GPIO output in push-pull mode */
+#define GPIO_PUSH_PULL          (0 << 1)
+
+/** Indicates single ended open drain mode (wired AND). */
+#define GPIO_LINE_OPEN_DRAIN    (1 << 2)
+/** Indicates single ended open source mode (wired OR). */
+#define GPIO_LINE_OPEN_SOURCE   (0 << 2)
+
+/** Configures GPIO output in open drain mode (wired AND).
+ *
+ * @note 'Open Drain' mode also known as 'Open Collector' is an output
+ * configuration which behaves like a switch that is either connected to ground
+ * or disconnected.
  */
-/** GPIO pin to be input. */
-#define GPIO_DIR_IN		(0 << 0)
-
-/** GPIO pin to be output. */
-#define GPIO_DIR_OUT		(1 << 0)
-
-/** @cond INTERNAL_HIDDEN */
-#define GPIO_DIR_MASK		0x1
-/** @endcond */
-/** @} */
-
-
-/**
- * @name GPIO interrupt flags
- * The `GPIO_INT_*` flags are used with `gpio_pin_configure` or `gpio_port_configure`,
- * to specify how input GPIO pins will trigger interrupts.
- * @{
+#define GPIO_OPEN_DRAIN         (GPIO_SINGLE_ENDED | GPIO_LINE_OPEN_DRAIN)
+/** Configures GPIO output in open source mode (wired OR).
+ *
+ * @note 'Open Source' is a term used by software engineers to describe output
+ * mode opposite to 'Open Drain'. It has no corresponding hardware schematic
+ * and is generally not used by hardware engineers.
  */
-/** GPIO pin to trigger interrupt. */
-#define GPIO_INT		(1 << 1)
+#define GPIO_OPEN_SOURCE        (GPIO_SINGLE_ENDED | GPIO_LINE_OPEN_SOURCE)
 
-/** GPIO pin trigger on level low or falling edge. */
-#define GPIO_INT_ACTIVE_LOW	(0 << 2)
+/** Enables GPIO pin pull-up. */
+#define GPIO_PULL_UP            (1 << 4)
```

Comment:
Bit 3 is defined by [Linux DT bindings](https://elixir.bootlin.com/linux/latest/source/include/dt-bindings/gpio/gpio.h) as:
```
/* Bit 3 express GPIO suspend/resume and reset persistence */
#define GPIO_PERSISTENT 0
#define GPIO_TRANSITORY 8
```
We should add support for this flag later. I updated issue description to clarify this. Bits 6, 7 are reserved in case Linux guys add more flags.

Thanks for the feedback! I fixed all the other issues you reported.

Reply:
Ah, I get it now; I didn't realize you were matching values a well as symbolic names. Thanks for explaining.

Reply:
> Thanks for the feedback! I fixed all the other issues you reported.

Just FYI, it took me a while to make sure that this was true, because the fix was squashed into f5857b7645 ("api: move non DTS GPIO flags away from DT bindings"), which was just supposed to be a copy/paste without changes type commit; I would have expected it in ("f64993a3ea api: gpio: Clean up GPIO dt-bindings flags")